codeit/services.js/ArticleService.js

```diff
@@ -0,0 +1,125 @@
+                    url.searchParams.append(key, params[key]); // new URL()로 만든 URL 객체 뒤에 파라미터에 들어가 있는 키와 값을 한 쌍의 쿼리 파라미터를 추가해주는 메서드
+                }
+            })
+
+            return fetch(url) // 리퀘스트 보냄, promise 리턴 받기 위해 return 키워드 사용
+
+            .then((res) => { // res는 서버로 부터 받은 리스폰스
+                if(!res.ok) { // 만약 리스폰스가 200번대가 아니면
+                    console.error('Article 불러오기 실패! 상태코드:', res.status) // 에러를 파라미터 안의 에러 메시지와 함께 출력. res.status는 리스폰스로 받은 상태코드를 반환함
+                    throw new Error ('Article 불러오기 실패!') // 에러를 발생시키고 코드를 catch문으로 넘김
+                }
+            return res.json() // 이 외의 경우에는 받은 리스폰스를 파싱 (else 생략 한 것)
+            })
+            .catch ((error)=> { // 리퀘스트 자체를 실패한 경우 & if 문에서 넘어온 throw 문 (상태코드 200이 아닌 경우) 경우만 실행됨
+                console.error (error.message) // 에러 메시지 출력
+            });
+    }, 
+
+
+    // 2. 게시글 상세 조회 : getArticle 함수
+    getArticle(articleId) { // 특정 게시글만 조회 할 수 있게 패스 파라미터 작성
+            return fetch(`${BASE_URL}/articles/${articleId}`) // 특정 게시글을 리퀘스트 요청
+            .then((res) => { // fetch Promise 에 리스폰스를 받으면 콜백 함수 실행
+                if (!res.ok) { // 리스폰스 받은 상태코드가 200번대가 아니면 실행
+                    console.error('Article 불러오기 실패! 상태코드:', res.status) // 에러를 파라미터 안의 에러 메시지와 함께 출력. res.status는 리스폰스로 받은 상태코드를 반환함
+                    throw new Error ('Article 불러오기 실패!') // 에러를 발생시키고 코드를 catch문으로 넘김
+                }
+                return res.json() // 이 외의 경우에는 받은 리스폰스를 파싱 (else 생략 한 것)
+            })
+            .then((data)=> { //윗줄의 파싱된 정보를 파라미터 안 data에 넣음
+                return data; // 파싱된 정보 반환
+            })
+            .catch ((error)=> { // 리퀘스트 자체를 실패한 경우 & if 문에서 넘어온 throw 문 (상태코드 200이 아닌 경우) 경우만 실행됨
+                console.error (error.message) // 에러 메시지 출력
+            })
+    }, 
+
+
+
+    // 3. 게시글 생성 : createArticle 함수
+    createArticle(articleData){ // 함수 선언
+        return fetch(`${BASE_URL}/articles`, { // 리퀘스트 요청 (첫번째 파라미터는 주소, 두 번짼느 옵션)
+            method: 'POST', // 메소드를 POST로 설정
+            body: JSON.stringify({ // 자바스크립트 객체를 JSON 문자열로 변환
+                title: articleData.title,
+                content: articleData.content,
+                image: articleData.image,
+            }), 
+            headers: { // 어떤 형식의 데이터를 보내는지 
+                'Content-Type': 'application/json' 
+            }, 
+        })
+            .then((res)=> { // res 는 서버에서 받은 리스폰스
+                if(!res.ok) { // 만약 리스폰스가 200번대가 아니라면 실행
+                    console.error('Article 불러오기 실패! 상태코드:', res.status) // 에러를 파라미터 안의 에러 메시지와 함께 출력. res.status는 리스폰스로 받은 상태코드를 반환함
+                    throw new Error ('Article 불러오기 실패!') // 에러를 발생시키고 코드를 catch문으로 넘김
+                } 
+                return res.json()  // 이 외의 경우에는 받은 리스폰스를 파싱 (else 생략 한 것)
+            })
+
+            .catch ((error)=> { // 리퀘스트 자체를 실패한 경우 & if 문에서 넘어온 throw 문 (상태코드 200이 아닌 경우) 경우만 실행됨
+                console.error (error.message) // 에러 메시지 출력
+            })
+        },
```

Comment:
fetch를 사용하는 구간들이 다 비슷한 코드들을 사용하고있는데, 공통 유틸리티를 만들어보시는 것은 어떨까요? 간단한 예제로는

```suggestion
      // 공통 fetch 함수
      fetchWithErrorHandling(url, options = {}, actionName = "작업") {
          return fetch(url, options)
              .then(res => {
                  if (!res.ok) {
                      throw new Error(`${actionName} 실패! 상태코드: ${res.status}`);
                  }
                  return res.json();
              })
              .catch(error => {
                  console.error(`${actionName} 중 오류:`, error.message);
                  throw error;
              });
      },

      createArticle(articleData) {
          const options = {
              method: "POST",
              headers: { "Content-Type": "application/json" },
              body: JSON.stringify(articleData)
          };

          return this.fetchWithErrorHandling(
              `${BASE_URL}/articles`,
              options,
              "Article 생성"
          );
      },
```